### PR TITLE
New Language: Ukrainian - Lang 32

### PR DESCRIPTION
**Pull request by pamidur**
#### Database Migration
NO

#### Description
Added support for Ukrainian language - made changes to parser and added tests.

### CODE REVIEW

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs [hunk 1]
```diff
@@ -198,6 +198,25 @@ public void should_parse_language_polish_subbed(string postTitle)
             result.Languages.Should().BeEquivalentTo(Language.Unknown);
         }
 
+        [TestCase("Movie.Title.1994.1080p.UA.XviD")]
+        [TestCase("Movie.Title.1994.1080p.UKR.XviD")]
+        [TestCase("Movie.Title.1994.1080p.2xUKR.XviD")]
```
Comment (bakerboy448) [inline src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:203]: are there true examples for `2xUKR` releases? nothing on PreDB

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs [hunk 2]
```diff
@@ -198,6 +198,25 @@ public void should_parse_language_polish_subbed(string postTitle)
...
+        public void should_parse_language_ukrainian(string postTitle)
+        {
+            var result = Parser.Parser.ParseMovieTitle(postTitle, true);
+
+            result.Languages.Should().BeEquivalentTo(Language.Ukrainian);
+        }
+
+        [TestCase("Movie.Title.1994.H.265.[UKR_ENG]")]
```
Comment (bakerboy448) [inline src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:211]: same as above; nothing on predb

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs [hunk 3]
```diff
@@ -198,6 +198,25 @@ public void should_parse_language_polish_subbed(string postTitle)
...
+        [TestCase("Movie.Title.1994.UKR/ENG.1080p")]
```
Comment (bakerboy448) [inline src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:212]: same as above; nothing on predb

File: src/NzbDrone.Core/Parser/LanguageParser.cs [hunk 4]
```diff
@@ -14,7 +14,7 @@ public static class LanguageParser
     {
         private static readonly Logger Logger = NzbDroneLogger.GetLogger(typeof(LanguageParser));
 
-        private static readonly Regex LanguageRegex = new Regex(@"(?:\W|_|^)(?<italian>\b(?:ita|italian)\b)|(?<german>\b(?:german|videomann|ger)\b)|(?<flemish>flemish)|(?<bulgarian>bgaudio)|(?<brazilian>dublado)|(?<greek>greek)|(?<french>(?:\W|_)(?:FR|VO|VFF|VFQ|VFI|VF2|TRUEFRENCH|FRE|FRA)(?:\W|_))|(?<russian>\brus\b)|(?<english>\beng\b)|(?<hungarian>\b(?:HUNDUB|HUN)\b)|(?<hebrew>\bHebDub\b)|(?<polish>\b(?:PL\W?DUB|DUB\W?PL|LEK\W?PL|PL\W?LEK)\b)|(?<chinese>\[(?:CH[ST]|BIG5|GB)\]|简|繁|字幕)",
+        private static readonly Regex LanguageRegex = new Regex(@"(?:\W|_|^)(?<italian>\b(?:ita|italian)\b)|(?<german>\b(?:german|videomann|ger)\b)|(?<flemish>flemish)|(?<bulgarian>bgaudio)|(?<brazilian>dublado)|(?<greek>greek)|(?<french>(?:\W|_)(?:FR|VO|VFF|VFQ|VFI|VF2|TRUEFRENCH|FRE|FRA)(?:\W|_))|(?<russian>\brus\b)|(?<english>\beng\b)|(?<hungarian>\b(?:HUNDUB|HUN)\b)|(?<hebrew>\bHebDub\b)|(?<polish>\b(?:PL\W?DUB|DUB\W?PL|LEK\W?PL|PL\W?LEK)\b)|(?<chinese>\[(?:CH[ST]|BIG5|GB)\]|简|繁|字幕)|(?<ukrainian>(?:\dx|_|\b)(?:UKR|UA)(?:_|\b))",
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Parser/LanguageParser.cs:17]: try `(?:\W|_|^)(?<italian>\b(?:ita|italian)\b)|(?<german>\b(?:german|videomann|ger)\b)|(?<flemish>flemish)|(?<bulgarian>bgaudio)|(?<brazilian>dublado)|(?<greek>greek)|(?<french>(?:\W|_)(?:FR|VO|VFF|VFQ|VFI|VF2|TRUEFRENCH|FRE|FRA)(?:\W|_))|(?<russian>\brus\b)|(?<english>\beng\b)|(?<hungarian>\b(?:HUNDUB|HUN)\b)|(?<hebrew>\bHebDub\b)|(?<polish>\b(?:PL\W?DUB|DUB\W?PL|LEK\W?PL|PL\W?LEK)\b)|(?<chinese>\[(?:CH[ST]|BIG5|GB)\]|简|繁|字幕)|(?<ukrainian>(?:(?:\dx)?UKR|UA))` instead

new `(?<ukrainian>(?:(?:\dx)?UKR|UA))`

old `(?<ukrainian>(?:\dx|_|\b)(?:UKR|UA)(?:_|\b))`

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
some changes to make and additional information on the examples needed.

The permissions were also incorrectly changed from 100644 to 100755; that needs to be corrected.

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs (hunk 1 above)
Comment (pamidur) [reply src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:203]: it is quite popular notation on private trackers. Others are much less frequent but sometimes they are there.

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs (hunk 2 above)
Comment (pamidur) [reply src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:211]: This notation although rare, sometimes appears on private trackers.

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs (hunk 3 above)
Comment (pamidur) [reply src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:212]: image

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs (hunk 1 above)
Comment (pamidur) [reply src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:203]: image

File: src/NzbDrone.Core/Parser/LanguageParser.cs (hunk 4 above)
Comment (pamidur) [reply src/NzbDrone.Core/Parser/LanguageParser.cs]: Thank you, works like a charm!

**Comment by pamidur** [comment]
Added info on examples, fixed file permissions, updated regex

**Comment by pamidur** [comment]
Hey guys, I'm sorry to bother you, but I think I have addressed all your considerations and recently rebased this PR on top of develop branch. Can you tell me please what prevents from accepting it?

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs (hunk 1 above)
Comment (bakerboy448) [reply src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:203]: these examples in jackett are TV nor movies nor do the test follow the naming schemas

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
conflicts & evidence of movies in the wild required

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs (hunk 1 above)
Comment (pamidur) [reply src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:203]: зображення
зображення
зображення

there are lots of releases that follow this naming (2xUkr, 3xUkr, etc) on trackers, it also happens something like 2xEng

What do you mean by "tests does not follow the schemas" ? Could you please elaborate, so I can improve my PR ?

**Comment by pamidur** [comment]
Fixed conflicts.
These are proofs for movies in the wild:
зображення
зображення

File: src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs (hunk 1 above)
Comment (bakerboy448) [reply src/NzbDrone.Core.Test/ParserTests/LanguageParserFixture.cs:208]: > "tests does not follow the schemas"

your tests don't reflect the actual release names being parsed.  

your test cases should be based on the release names in the wild (but with their titles replaced to movie title or similar

**Review by bakerboy448 — CHANGES_REQUESTED** [review]

**Comment by pamidur** [comment]
made test based on real names in the wild

**Pull request closed without merging**